python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -68,6 +80,17 @@ def __init__(self, registryConfig, schemaConfig):
         self._engine = create_engine(self.config['db'])
         self._schema.metadata.create_all(self._engine)
         self._datasetTypes = {}
+        self._connection = self._engine.connect()
+
+    @contextlib.contextmanager
+    def transaction(self):
+        trans = self._connection.begin()
+        try:
+            yield
+            trans.commit()
```

Comment:
Should the commit be in a `finally` block?

Reply:
No, because we don't want to commit in case an exception is raised.